react-client/src/Components/App.js

```diff
@@ -42,59 +41,79 @@ function App() {
     return () => {};
   }, []);
 
-  useEffect(() => {
-    return () => {};
-  }, []);
-
   const injectDOMEventInBody = async () => {
     document.addEventListener("click", (e) => {
       if (e.target.id !== "closedompeeker" && domSwitch == true) {
         e.preventDefault();
 
-        const children = [...e.target.children].map(child => {
-          return { id: child.id, class: child.className };
+        const children = [...e.target.children].map((child) => {
+          return {
+            id: child.id.trim() !== "" ? "#" + child.id : null,  class: child.className.trim() !== "" ? "." + child.className : null, childtag: child.localName
+          };
         });
 
+        if (e.target.id.trim() !== "") {
+          var eltarget = document.querySelector(`#${e.target.id}`);
+        } else {
+          var eltarget = document.querySelector(`.${e.target.className}`);
+        }
+        var fontsize = window
+          .getComputedStyle(eltarget, null)
+          .getPropertyValue("font-size");
+
+        var fontfamily = window
+          .getComputedStyle(eltarget, null)
+          .getPropertyValue("font-family");
+
+        var fontcolor = window
+          .getComputedStyle(eltarget, null)
+          .getPropertyValue("color");
+
+        var rgbArr = fontcolor.substring(4).slice(0, -1).split(",");
+
+        var colorhex =
```

Comment:
I feel this code can be shorten further. This is redundant.

Reply:
You can probably use `reduce()` to reduce the clutter on this code.

Reply:
You can also do the following code to reduce the code: `rgbArr.reduce((init, curr) => (init += parseInt(curr).toString(16)), '#')`

Reply:
noted already implmented